### PR TITLE
[DotNet][DateTime] Refactor GetModAndDate method since it has too many parameters

### PR DESCRIPTION
**Pull request by ceciliaavila**
- Created new class ModAndDateResult containing the properties we need to return in the method GetModAndDate
- Redesigned GetModAndDate method to remove the out/ref parameters and return an instance of a ModAndDateResult object.

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Text.DateTime/Utilities/GetModAndDateResult.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using DateObject = System.DateTime;
+
+namespace Microsoft.Recognizers.Text.DateTime.Utilities
+{
+    public class GetModAndDateResult
```
Comment (VictorGrycuk) [inline .NET/Microsoft.Recognizers.Text.DateTime/Utilities/GetModAndDateResult.cs]: I would like to suggest `ModAndDateResult` as name for the class. Usually only non void methods start with `Get`.

Comment (ceciliaavila) [reply .NET/Microsoft.Recognizers.Text.DateTime/Utilities/GetModAndDateResult.cs]: Feedback applied!

**Comment by ParadoxARG** [comment]
I would suggest removing the [Recognizers-Text] from the title, it's a redundant label as you are publishing on the Recognizers-Text repo.

**Comment by ceciliaavila** [comment]
> I would suggest removing the [Recognizers-Text] from the title, it's a redundant label as you are publishing on the Recognizers-Text repo.

It's true. Thanks!

File: .NET/Microsoft.Recognizers.Text.DateTime/Utilities/GetModAndDateResult.cs (hunk 1 above)
Comment (JuanAr) [reply .NET/Microsoft.Recognizers.Text.DateTime/Utilities/GetModAndDateResult.cs]: I think @VictorGrycuk referred to rename only the class. You also renamed the method, which I'd roll back.

Comment (JuanAr) [reply .NET/Microsoft.Recognizers.Text.DateTime/Utilities/GetModAndDateResult.cs]: `public class ModAndDateResult` and `public ModAndDateResult GetModAndDate(`

Comment (ceciliaavila) [reply .NET/Microsoft.Recognizers.Text.DateTime/Utilities/GetModAndDateResult.cs]: Done!

**Review by JuanAr — APPROVED** [review]
👍

**Pull request closed without merging**